articles/node.js-vs-python/index.md

```diff
@@ -0,0 +1,123 @@
+
+Before we look at the comparison of the two technology, let's first look at the primary differences between the two.
+
+Node.js is an open-source, cross-platform technology and Javascript runtime environment capable of executing javascript at the server-side. It depends on Chrome's V8 JavaScript engine. This engine was designed by Google to compile JavaScript functions into machine code.
+
+On the other hand, Python is a general-purpose programming language used in web development to create dynamic websites using frameworks like Flask, Django, and Pyramid. For the most part, Python runs on Google's Apps Engine. Likewise created by Google, the App Engine allows you to create web applications with Python and permits you to benefit from various libraries that the best Python developers use.
+
+To make a good decision on which technology to use, we will  consider the following criteria:
+1. Speed
+2. Scalability
+3. Universality
+4. Extensibility
+5. Architecture
+
+### SPEED
+Performance and Speed are the main features that every developer wishes his project to have at the end. Every developer aims to deliver a high-performance application to the user.
+
+#### Node.Js
+Since Node.js uses [V8 Engine](https://en.wikipedia.org/wiki/V8_(JavaScript_engine)) to interpret JavaScript code, its performance is remarkable. V8 is a powerful JavaScript engine that is provided by Google. It converts JavaScript code into machine code that will be understood by the processor.
+
+Node.js is also very responsive and has a very high loading speed.
+
+#### Python
+Unlike Java which is a compiled language, Python is an interpreted language. It is generally slower than the compiled languages. This makes Python lose to Node.js in terms of performance.
+
+Python is not suitable for apps that require high execution speed. This is because of the single flow of code in Python which leads to slow processing of requests. Python web applications are therefore slower.
+
+### SCALABILITY
+Scalability is the ability of a web app to adapt to changes like an increase in performance, new features, etc.
+
+#### Node.Js
+Unlike Python, Node.js uses different modules for its functionality. Therefore, for the increase of scalability of an app developed by Node.js few modules are added. This will increase the demand for using Node.js by developers.
+
+Node.js is an asynchronous technology. (can run multiple processes)Hence it facilitates multithreading.
+#### Python
+Python does not support multithreading. Therefore, scalability is not easy. For Python to have easy scalability, libraries have to be used. However, this does not mean that it can compete with Node.js in terms of scalability.
+
+### UNIVERSALITY
+
+ #### Node.Js
+Node.js is used in backend development. For frontend development, developers prefer JavaScript. This saves on resources since only one team with Node.js skills is hired. This feature of Node.js has, in turn, led many developers to choose it.
```

Comment:
Please expound on this point on saving resources and hiring one team as it's not that clear.